core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -170,6 +182,138 @@ export class ContributionAndReviewService {
     );
   }
 
+  async fetchTranslationSuggestionsAsync(
+      explorationId: string
+  ): Promise<FetchSuggestionsResponse> {
+    const explorationBackendResponse = await this.
+      readOnlyExplorationBackendApiService.fetchExplorationAsync(
+        explorationId, null);
+    return (
+      this.contributionAndReviewBackendApiService.
+        fetchSuggestionsAsync(
+          'REVIEWABLE_TRANSLATION_SUGGESTIONS',
+          null,
+          0,
+          AppConstants.SUGGESTIONS_SORT_KEY_DATE,
+          explorationId
+        ).then((fetchSuggestionsResponse) => {
+          const exploration: Exploration = this.explorationObjectFactory.
+            createFromExplorationBackendResponse(
+              explorationBackendResponse);
+          const sortedTranslationSuggestions = (
+            this.sortTranslationSuggestionsByState(
+              fetchSuggestionsResponse.suggestions,
+              exploration.getStates(),
+              exploration.initStateName));
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const responseSuggestionIdToDetails: SuggestionDetailsDict = {};
+          sortedTranslationSuggestions.forEach((suggestion) => {
+            const suggestionDetails = {
+              suggestion: suggestion,
+              details: (
+                fetchSuggestionsResponse.target_id_to_opportunity_dict[
+                  suggestion.target_id])
+            };
+            responseSuggestionIdToDetails[
+              suggestion.suggestion_id] = suggestionDetails;
+          });
+          return {
+            suggestionIdToDetails: responseSuggestionIdToDetails,
+            more: false
+          };
+        })
+    );
+  }
+
+  sortTranslationSuggestionsByState(
+      translationSuggestions: SuggestionBackendDict[],
+      states: States,
+      initStateName: string | null
+  ): SuggestionBackendDict[] {
+    if (!initStateName) {
+      return translationSuggestions;
+    }
+
+    const stateNamesInOrder = this.computeGraphService.
+      computeBfsTraversalOfStates(
+        initStateName,
+        states,
+        initStateName
+      );
+    const translationSuggestionsByState = (
+      ContributionAndReviewService
+        .groupTranslationSuggestionsByState(translationSuggestions));
+    const sortedTranslationCards: SuggestionBackendDict[] = [];
+
+    for (const stateName of stateNamesInOrder) {
+      const cardsForState = (
+        translationSuggestionsByState.get(stateName) || []);
+      cardsForState.sort(
+        ContributionAndReviewService.compareTranslationSuggestions);
+      sortedTranslationCards.push(...cardsForState);
+    }
+    return sortedTranslationCards;
+  }
+
+  static groupTranslationSuggestionsByState(
```

Comment:
These functions below can also be made `private` right? In general, we should keep the visibility as low as possible.

Reply:
Done, Thanks.